File: api/models/donante.py
"""Esquemas del donante — ARCHIVO COMPARTIDO.

Los consumen los blueprints de donantes, resumen y reportes.
Avisa al equipo antes de cambiar los campos.
"""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class Donante(BaseModel):
    id: int
    nombre: str
    segmento: str  # "en_riesgo", "alto_valor"
    estado: str    # "activo", "inactivo"

    montoTotal: Decimal
    ultimaDonacion: datetime
    primeraDonacion: datetime
    frecuencia: str
    montoPromedio: Decimal
    acumulado12Meses: Decimal
    detalleEstado: str
    pagos: list["PagoDonante"] = Field(default_factory=list)
    llamadas: list["LlamadaDonante"] = Field(default_factory=list)


class Donativo(BaseModel):
    id: int
    donante_id: int
    monto: Decimal
    fecha: datetime
    estatus: str  # "cobrado", "rechazado", "pendiente"


class PagoDonante(BaseModel):
    id: int
    date: datetime
    amount: Decimal
    status: str


class LlamadaDonante(BaseModel):
    id: int
    date: datetime
    result: str
    notes: str

File: ios/SistemaIngresos/Core/Model/Donante.swift
import Foundation

struct Donante: Identifiable, Codable {
    let id: Int
    let nombre: String
    let segmento: String
    let estado: String
    let nivelRiesgo: NivelRiesgoDonante
    var montoTotal: Decimal = 0
    var ultimaDonacion: Date = .distantPast
    var primeraDonacion: Date = .distantPast
    var frecuencia: String = ""
    var montoPromedio: Decimal = 0
    var acumulado12Meses: Decimal = 0
    var detalleEstado: String = ""
    var pagos: [PagoDonante] = []
    var llamadas: [LlamadaDonante] = []

    var estadoVisual: EstadoVisualDonante {
        EstadoVisualDonante(nivelRiesgo: nivelRiesgo)
    }
}

enum NivelRiesgoDonante: String, Codable {
    case verde
    case amarillo
    case naranja
    case rojo
}

enum EstadoVisualDonante {
    case verde, amarillo, naranja, rojo

    init(nivelRiesgo: NivelRiesgoDonante) {
        switch nivelRiesgo {
        case .verde: self = .verde
        case .amarillo: self = .amarillo
        case .naranja: self = .naranja
        case .rojo: self = .rojo
        }
    }

    var title: String {
        switch self {
        case .verde: "Riesgo bajo"
        case .amarillo: "Riesgo medio"
        case .naranja: "Riesgo elevado"
        case .rojo: "Riesgo alto"
        }
    }
}

struct PagoDonante: Identifiable, Codable {
    let id: Int
    let date: Date
    let amount: Decimal
    let status: EstadoPagoDonante
}

enum EstadoPagoDonante: String, Codable {
    case paid = "cobrado"
    case rejected = "rechazado"
    case pending = "pendiente"
    var title: String {
        switch self {
        case .paid: "Cobrado"
        case .rejected: "Rechazado"
        case .pending: "Pendiente"
        }
    }
}

struct LlamadaDonante: Identifiable, Codable {
    let id: Int
    let date: Date
    let result: String
    let notes: String
}
